### PR TITLE
Added JavaScript/NodeJS examples

### PR DESCRIPTION
**Pull request by leonardobagi**
Added both Vanilla JS and Node code examples for the docs

### CODE REVIEW

**Comment by leonardobagi** [comment]
Thanks for reminding.

Me and some boys are working on `geometry-dash.js` and I found it interesting to add some examples to it. And this in fact could work at least in NodeJS with Axios, because there you can easily empty your `User-Agent` header, but I don't think that would be possible in Vanilla JS.

In addition, Axios automatically handles the `Content-Type` header so no worries there. I have never tried to make a request from the browser, but I am 100% sure it can be done in NodeJS (as it is working on `geometry-dash.js`).

Later, (aka after my class) I will add the needed code for making the request from Axios and try it on browser, if it does not work, we can try Axios for the web, if it doesn't work either, just entirely remove Vanilla JS from the examples

**Comment by Wyliemaster** [comment]
Not sure if you are aware, but when the user agent is made empty, axios uses the default user agent which is the version of axios you are using.
I've found that using `axios.defaults.headers.common["User-Agent"] = "";` solves this issue

**Comment by matcool** [comment]
You can't make requests to gd from the browser without using some sort of proxy due to CORS (Cross Origin Resource Sharing)

**Comment by leonardobagi** [comment]
Gotta remove the Vanilla JS examples then, it won't work unless the user has managed to disable CORS or use `no-cors` mode, which will probably not work to retrieve data

**Comment by SMJSGaming** [comment]
Why the excessive use of external libraries tho, I'd personally recommend using the already present http library to handle this stuff

**Comment by leonardobagi** [comment]
I would use https to make the examples, but on second thought, making it would be pretty much useless as the majority uses `axios` or something like `node-fetch` to make HTTP requests. Tho I could use `XMLHttpRequest` as well but that would be way too complicated. We could also use the `fetch` thing which is coming built-in in NodeJS soon, but I still don't think it is a very good ideia to rely on it too soon, as it's experimental and I also don't know how the headers will be set there.

**Comment by leonardobagi** [comment]
There is no reason in making it use https since a relatively small amount of people use it

**Comment by Cvolton** [comment]
boomlings.com doesn't support https to begin with (on this note, the www is also required)

**Comment by leonardobagi** [comment]
the https package? I think it's as simple as removing the S on the protocol

**Comment by SMJSGaming** [comment]
I said `http` not `https`, also, fetch hasn't been experimental for ages now, besides that it's a common practice to avoid relying too much on libraries and this here is a good example where you should stop and think if it's truly worth installing something external if it already exists internally.

**Comment by leonardobagi** [comment]
> I will implement the `fetch` API for both Javascript and NodeJS, as it has already replaced `node-fetch` it is the best way for now 

Attending to @SMJSGaming

File: docs/topics/encryption/base64.md [hunk 1]
```diff
@@ -30,4 +30,19 @@ def base64_decode(string: str) -> str:
     return base64.urlsafe_b64decode(string.encode()).decode()
 ```
 
+### **Javascript**
+
+```js
+const base64Encode = string => btoa(string).replace("+", "-").replace("/", "_");
+
```
Comment (SMJSGaming) [inline docs/topics/encryption/base64.md:37]: Why are you using callbacks, these are plain functions

File: scripts/installPackages.js [hunk 2]
```diff
@@ -1,14 +1,20 @@
...
 // made by Homura
+// refactored by LeonardoBagi
 const chalk = require("chalk");
-const fs = require("fs");
 const childProcess = require("child_process");
 
 const package = require("../package.json");
-const missing = Object.keys(package.dependencies)
-    .filter((package) => !fs.existsSync(`../node_modules/${package}`));
+const missing = Object.keys(package.dependencies).filter(package => {
```
Comment (SMJSGaming) [inline scripts/installPackages.js:8]: I don't see how a brute forced try/catch is more efficient than checking if the module exists

File: scripts/installPackages.js [hunk 3]
```diff
@@ -1,14 +1,20 @@
...
+        return true;
+    }
+});
 
 if (missing.length > 0) {
     console.log(chalk.hex("#79b0fc")("Installing Packages: ") + missing.join(", "));
-    childProcess.exec(`npm i --save ${missing.join(" ")}`);
-}
\ No newline at end of file
+    childProcess.execSync(`npm i ${missing.join(" ")}`);
```
Comment (naoei) [inline scripts/installPackages.js]: These changes are definitely out of what we're trying to add with this PR. I'd suggest moving them into another PR or just removing them entirely. This also includes the changes done with `package.json`, and `generator.js`

**Review by naoei — CHANGES_REQUESTED** [review]

File: docs/topics/encryption/base64.md (hunk 1 above)
Comment (leonardobagi) [reply docs/topics/encryption/base64.md:37]: I thought it was clearer to just use arrow functions in this case because it's a simple one line thing

File: scripts/installPackages.js (hunk 2 above)
Comment (leonardobagi) [reply scripts/installPackages.js:8]: That other code was always saying that the package didn't exist tho.. `require.resolve()` will only return the file name of the module so it's better to check it via that than to look into `node_modules` as far as I know

**Comment by leonardobagi** [comment]
> These changes are definitely out of scope with what we're trying to do in this PR. I'd suggest moving them into another PR or just removing them entirely. This also includes the changes done with `package.json`, and `generator.js`

Reverted those and I shall open another PR later containing changes and also taking what SMJS said to improve my code

**Comment by nekitdev** [comment]
I'm still not sure we need examples beyond Python, though. Mainly because the examples more or less explain logic, and imo showing an example that you can run is more comfy than writing pseudocode.

**Comment by leonardobagi** [comment]
Some things are done differently (like hashing and stuff) in different languages. These examples were mainly created to help with that

**Comment by yukisnow0** [comment]
it's a given whether or not something can be done differently in an entirely different platform, and regardless i find it useless to consider stuff like that when the average programmer should be able to transcribe something in a different language to the one that they use; documentations commonly use python because it's the nicest language that looks good as pseudocode due to its immense simplicity and ease of use
